comp/dogstatsd/listeners/connections_tracker.go

```diff
@@ -115,7 +120,11 @@ func (t *ConnectionTracker) HandleConnections() {
 					// We don't have a choice, setting a 0 timeout would likely be a retryable error.
 					err = c.Close()
 				}
-				log.Debugf("dogstatsd-%s: failed to shutdown connection: %v", t.name, err)
+
+				if err != nil {
+					log.Debugf("dogstatsd-%s: failed to shutdown connection (read): %v", t.name, err)
+					err = nil
+				}
```

Comment:
Wait, am I totally missing something? Why do we do the exact same loop _twice_? (L93-107, and then L111-128)

Reply:
Doh, I can't read. :)